x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table_header.tsx

```diff
@@ -151,20 +252,171 @@ const SearchField = ({
           fullWidth
         />
       </EuiFlexItem>
-      <CurrentPageOfTotal pageSize={pageSize} total={totalRulesCount} />
+      <CurrentPageOfTotal
+        pageSize={pageSize}
+        total={totalRulesCount}
+        selectedRules={selectedRules}
+        refetchRulesStates={refetchRulesStates}
+        setSelectAllRules={setSelectAllRules}
+        setSelectedRules={setSelectedRules}
+      />
     </div>
   );
 };
 
-const CurrentPageOfTotal = ({ pageSize, total }: RuleTableCount) => (
-  <EuiFlexItem grow={true}>
-    <EuiSpacer size="xl" />
-    <EuiText size="xs" textAlign="left" color="subdued" style={{ marginLeft: '8px' }}>
-      <FormattedMessage
-        id="xpack.csp.rules.rulesTable.showingPageOfTotalLabel"
-        defaultMessage="Showing {pageSize} of of {total, plural, one {# rule} other {# rules}}"
-        values={{ pageSize, total }}
-      />
-    </EuiText>
-  </EuiFlexItem>
-);
+const CurrentPageOfTotal = ({
+  pageSize,
+  total,
+  selectedRules,
+  refetchRulesStates,
+  setSelectAllRules,
+  setSelectedRules,
+}: RuleTableCount) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const onPopoverClick = () => {
+    setIsPopoverOpen((e) => !e);
+  };
+
+  const postRequestChangeRulesStatus = useChangeCspRuleState();
+  const closePopover = () => setIsPopoverOpen(false);
+  const changeRulesStatus = async (status: 'mute' | 'unmute') => {
+    const bulkSelectedRules: RuleStateAttributesWithoutStatus[] = selectedRules.map(
+      (e: CspBenchmarkRulesWithStates) => ({
+        benchmark_id: e?.metadata.benchmark.id,
+        benchmark_version: e?.metadata.benchmark.version,
+        rule_number: e?.metadata.benchmark.rule_number!,
+        rule_id: e?.metadata.id,
+      })
+    );
+    // Only do the API Call IF there are no undefined value for rule number in the selected rules
+    if (!bulkSelectedRules.some((e) => e.rule_number === undefined)) {
+      await postRequestChangeRulesStatus(status, bulkSelectedRules);
+      await refetchRulesStates();
+      await closePopover();
+    }
+  };
+  const changeCspRuleStatusMute = async () => {
+    changeRulesStatus('mute');
+  };
+  const changeCspRuleStatusUnmute = async () => {
+    changeRulesStatus('unmute');
+  };
```

Comment:
i think both can be a single function that takes `mute` or `unmute` as an argument

Reply:
If i make it this way doesn't that mean I would need to do something like
 ```
onClick={() => {
        changeRulesState('unmute');
        setSelectedRules([]);
      }}
```
I wrapped those line to prevent calling the function in line (which looks kinda messy)